presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -457,7 +457,7 @@ private void destroyIfNecessary()
         }
     }
 
-    private static ListenableFuture<?> isBlocked(Operator operator)
+    private static ListenableFuture<?> getBlockedFuture(Operator operator)
     {
         ListenableFuture<?> blocked = operator.isBlocked();
         if (blocked.isDone()) {
```

Comment:
It looks like what you really want is to say, "I'm blocked" until both of these conditions are satisfied.  If so, you can simple replace the original method with:
```java
        return Futures.allAsList(operator.isBlocked(), operator.getOperatorContext().isWaitingForMemory());
```